### PR TITLE
[doc] Update docs/deprecated.md for 1.13

### PR DESCRIPTION
**Pull request by AkihiroSuda**
CHANGELOG is tracked in https://github.com/docker/docker/pull/28275

cc @vieux 

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

### CODE REVIEW

**Comment by vieux** [comment]
@AkihiroSuda already needs a rebase

File: docs/deprecated.md [hunk 1]
```diff
@@ -43,6 +43,32 @@ The daemon is moved to a separate binary (`dockerd`), and should be used instead
 Duplicate keys with conflicting values have been deprecated. A warning is displayed
 in the output, and an error will be returned in the future.
 
+### `MAINTAINER` in Dockerfile
```
Comment (AkihiroSuda) [inline docs/deprecated.md]: this section is just moved from the tail of the original file

File: docs/deprecated.md [hunk 2]
```diff
@@ -43,6 +43,32 @@ The daemon is moved to a separate binary (`dockerd`), and should be used instead
...
+### API calls without a version
+**Deprecated In Release: [v1.13](https://github.com/docker/docker/releases/)**
+
+**Target For Removal In Release: v1.16**
+
+API versions should be supplied to all API calls to ensure compatibility with
+future Engine versions. Instead of just requesting, for example, the URL
+`/containers/json`, you must now request `/v1.25/containers/json`.
+
+### Backing filesystem without `d_type` support for overlay/overlay2
```
Comment (AkihiroSuda) [inline docs/deprecated.md]: this section is new in this PR

**Comment by AkihiroSuda** [comment]
sorry, rebased :sweat_smile:  @vieux

**Comment by vieux** [comment]
ping @thaJeztah

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Pull request merged**